torch/distributed/elastic/rendezvous/api.py

```diff
@@ -86,7 +86,7 @@ def build(
         """
         # TODO swap to collectives comms API
         if rank == 0:
-            addr = local_addr or socket.getfqdn()
+            addr = local_addr or socket.gethostbyname(socket.getfqdn())
```

Comment:
This will break all IPv6 usages of torchelastic. gethostbyname only supports ipv4 so on IPv6 only hosts it ends up returning the localhost address (127.0.0.*)

We could try supporting this via getaddrinfo instead (which is probably the right solution) -- but since that returns multiple IP addresses we would need a much smarter way of propagating multiple addresses and deciding which one is routable

Reply:
Yes, you are right, `gethostbyname` not work on IPv6-only environments.
I've provide another version base on your comment, while would you think returning the first non-loopback ip will be enough on rendezvous phase ?